src/Twig/Component/Banner.php

```diff
@@ -0,0 +1,34 @@
+<?php
+
+namespace EasyCorp\Bundle\EasyAdminBundle\Twig\Component;
+
+use EasyCorp\Bundle\EasyAdminBundle\Config\Enum\BannerVariant;
+use Symfony\Component\OptionsResolver\OptionsResolver;
+use Symfony\UX\TwigComponent\Attribute\PreMount;
+
+class Banner
+{
+    public BannerVariant|string $variant = BannerVariant::Info;
+    public bool $withDismissButton = false;
+
+    #[PreMount]
+    public function preMount(array $data): array
+    {
+        $resolver = new OptionsResolver();
+        $resolver->setIgnoreUndefined(true);
+
+        $resolver->define('variant')
+            ->allowedTypes('string', BannerVariant::class)
+            ->allowedValues(...array_map(static fn(BannerVariant $variant): string => $variant->value, BannerVariant::cases()));
+
+        $resolver->define('withDismissButton')
+            ->allowedTypes('bool');
+
+        return $resolver->resolve($data) + $data;
+    }
+
+    public function getVariantCssClass(): string
```

Comment:
Import `Symfony\UX\TwigComponent\Attribute\ExposeInTemplate`
```suggestion
    #[ExposeInTemplate]
    public function getVariantCssClass(): string
```
so in twig you can simply access as `variantCssClass`

Reply:
I think it's a bit of a personal preference...

Personally I find the use of `#[ExposeInTemplate]` a bit confusing since it allows you to reference a Component property/method without using `this` keyword, and could be confused with public properties or Twig variables.

Reply:
I'm divided too, so for now I've decided to not use this `#[ExposeInTemplate]` attribute. But, I'll keep it mind for future uses. Thanks!